securitytxt-signer.sh

```diff
@@ -96,8 +100,8 @@ else
   GREPABLE_KEY=${KEY//0x/}
   FP=$(echo "$KEY_INFO" | grep -i "$GREPABLE_KEY" | sed -e 's/[^A-F0-9]//g')
   KEY="0x${FP}"
-  if ! [[ "${KEY}" = "0x${GREPABLE_KEY^^}" ]]; then
```

Comment:
Do you know why this didn't work on macOS?

Reply:
The error is: `securitytxt-signer.sh: line 104: \033[0;33mEXPANDED 0x${GREPABLE_KEY^^} TO ${KEY}\033[0;0m: bad substitution`

Maybe a problem with my bash:

```
GNU bash, version 3.2.57(1)-release (arm64-apple-darwin23)
Copyright (C) 2007 Free Software Foundation, Inc.
```

Reply:
That Bash is ancient.
```
GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2020 Free Software Foundation, Inc.
```

Reply:
This script already uses `sed` & `awk` and this uppercase conversion can be done with both. Despite they are all basic UNIX CLI tools I generally prefer keeping the set of commands minimal.

Reply:
This `${variable^^}` shell expansion was introduced in Bash 4.0. From the changelog:

```
hh. There are new case-modifying word expansions: uppercase (^[^]) and
    lowercase (,[,]).  They can work on either the first character or
    array element, or globally.  They accept an optional shell pattern
    that determines which characters to modify.  There is an optionally-
    configured feature to include capitalization operators.
 ```